app/src/ui/changes/changes-list.tsx

```diff
@@ -594,10 +594,16 @@ export class ChangesList extends React.Component<
   }
 
   private onStashEntryClicked = () => {
-    if (this.props.isShowingStashEntry) {
-      this.props.dispatcher.selectWorkingDirectoryFiles(this.props.repository)
+    const { isShowingStashEntry, dispatcher, repository } = this.props
+
+    if (isShowingStashEntry) {
+      dispatcher.selectWorkingDirectoryFiles(repository)
```

Comment:
these sequential, un-`await`-ed calls to dispatcher gave me pause here (in both the `if` and the `else` statements here) due to possible race conditions but i think its ok since it seems the work done by each of these methods is independent of one another.

it would be nice to consolidate these into a single `dispatcher`/`appStore` method for each case so the component can still "fire and forget" calls to dispatcher here, but i don't think its blocking

Reply:
:thumbsup: let's follow up after the beta ship today and see if we can settle on something we're happier with